src/Http/Controllers/Character/ContactsController.php

```diff
@@ -22,24 +22,119 @@
+            $character_ids = $user_group;
+
+        $contacts = $this->getCharacterContacts($character_ids);
+
+        return DataTables::of($contacts)
+            ->editColumn('name', function ($row) {
+
+                $character_id = $row->character_id;
+
+                if($row->contact_type === 'character'){
+                    $character = CharacterInfo::find($row->contact_id) ?: $row->contact_id;
+
+                    return view('web::partials.character', compact('character', 'character_id'));
+                }
+
+                if($row->contact_type === 'corporation'){
+                    $corporation = CorporationInfo::find($row->contact_id) ?: $row->contact_id;
+
+                    return view('web::partials.corporation', compact('corporation', 'character_id'));
+                }
+
+                return view('web::partials.unknown', [
+                    'unknown_id' => $row->contact_id,
+                    'character_id' => $character_id,
+                    ]);
+            })
+            ->editColumn('label_ids', function ($row) {
+
+                if(isset($row->label_ids)) {
+                    $labels = $this->getCharacterContactLabels($row->character_id);
+
+                    return $labels->whereIn('label_id', $row->label_ids)->implode('label_name', ', ');
+                }
+
+                return '';
+            })
+            ->addColumn('standing_view', function ($row) {
+
+                if($row->standing > 0)
+                    return "<b class='text-success'>" . $row->standing . '</b>';
+
+                if($row->standing < 0)
+                    return "<b class='text-danger'>" . $row->standing . '</b>';
+
+                return '<b>' . $row->standing . '</b>';
+
+            })
+            ->addColumn('links', function ($row) {
+
+                return view('web::partials.links', ['id' => $row->contact_id, 'type' => $row->contact_type]);
+            })
+            ->addColumn('is_in_group', function ($row) use ($user_group) {
+
+                return $user_group->intersect(collect($row->contact_id))->isNotEmpty();
+            })
+            ->filterColumn('name', function ($query, $keyword) {
+                $resolved_ids = ResolvedIds::where('name', 'like', '%' . $keyword . '%')->get()->map(function ($resolved_id) { return $resolved_id->id; });
+                $character_info_ids = CharacterInfo::where('name', 'like', '%' . $keyword . '%')->get()->map(function ($character_info) { return $character_info->character_id; });
+                $corporation_info_ids = CorporationInfo::where('name', 'like', '%' . $keyword . '%')->get()->map(function ($corporation_info) { return $corporation_info->corproation_id; });
+
+                $query->whereIn('contact_id', array_merge($resolved_ids->toArray(), $character_info_ids->toArray(), $corporation_info_ids->toArray()));
+            })
+            ->filterColumn('label_ids', function ($query, $keyword) {
```

Comment:
You can replace this by the one bellow which will return much proper results and cover all labels

```
->filterColumn('label_ids', function ($query, $keyword) use ($character_ids) {

	$labels = CharacterContactLabel::where('label_name', 'like', '%' . $keyword . '%')
		->whereIn('character_id', $character_ids)
		->get();

	foreach ($labels as $label)
		$query->whereRaw(DB::raw('JSON_CONTAINS(label_ids, ' . $label->label_id . ')'));
})
```